crates/oxc_linter/src/rules/react/jsx_no_duplicate_props.rs

```diff
@@ -0,0 +1,117 @@
+use oxc_ast::{
+    ast::{JSXAttributeItem, JSXAttributeName},
+    AstKind,
+};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::{self, Error},
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{Atom, Span};
+use rustc_hash::FxHashMap;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error(
+    "eslint-plugin-react(jsx-no-duplicate-props): No duplicate props allowed. The prop \"{0}\" is duplicated."
+)]
+#[diagnostic(
+    severity(warning),
+    help("Remove one of the props, or rename them so each prop is distinct.")
+)]
+struct JsxNoDuplicatePropsDiagnostic(Atom, #[label] pub Span, #[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct JsxNoDuplicateProps;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// This rule prevents duplicate props in JSX elements.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// Having duplicate props in a JSX element is most likely a mistake.
+    /// Creating JSX elements with duplicate props can cause unexpected behavior in your application.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// // Bad
+    /// <App a a />;
+    /// <App foo={2} bar baz foo={3} />;
+    ///
+    /// // Good
+    /// <App a />;
+    /// <App bar baz foo={3} />;
+    ///
+    /// ```
+    JsxNoDuplicateProps,
+    correctness
+);
+
+impl Rule for JsxNoDuplicateProps {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let AstKind::JSXOpeningElement(jsx_opening_elem) = node.kind() else { return };
+
+        let mut props: FxHashMap<Atom, Span> = FxHashMap::default();
+
+        for attr in &jsx_opening_elem.attributes {
+            let JSXAttributeItem::Attribute(jsx_attr) = attr else { continue };
+
+            let JSXAttributeName::Identifier(ident) = &jsx_attr.name else { continue };
+
+            if let Some(old_span) = props.insert(ident.name.clone(), jsx_attr.span) {
+                ctx.diagnostic(JsxNoDuplicatePropsDiagnostic(
+                    ident.name.clone(),
+                    old_span,
+                    jsx_attr.span,
+                ));
+            }
+        }
```

Comment:
not sure whether `FxHashMap` is the best choice here.

And of re-inserting and checking if `some` was returned is a good choice - happy to adjust

Reply:
This is good 👍 All the other redeclaration checks are done this way.

Reply:
:+1: thanks